### PR TITLE
8342827: Fix order of @param tags in other modules

### PR DESCRIPTION
**Pull request by hns**
Please review a doc-only change to fix the order of javadoc @param tags in in various OpenJDK modules. This is the third and last PR to fix the order of @param tags in OpenJDK libraries.

We are working on a javadoc feature to add an opt-in doclint warning for @param tags that don't match the order of parameters in the documented element. The warning will be enabled in OpenJDK builds and covers all uses of the @param tag, i.e. parameters of executable elements, type parameters, and record components.

I compared the generated API docs built with this branch with API docs built from master branch to make sure they are identical.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8342827](https://bugs.openjdk.org/browse/JDK-8342827): Fix order of @param tags in other modules (**Sub-task** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**) Review applies to [70c1d49f](https://git.openjdk.org/jdk/pull/21637/files/70c1d49f6c220af249f3c4e4ade14195c3ed32da)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)
 * [Iris Clark](https://openjdk.org/census#iris) (@irisclark - **Reviewer**)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)
 * [Joe Darcy](https://openjdk.org/census#darcy) (@jddarcy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21637/head:pull/21637` \
`$ git checkout pull/21637`

Update a local copy of the PR: \
`$ git checkout pull/21637` \
`$ git pull https://git.openjdk.org/jdk.git pull/21637/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21637`

View PR using the GUI difftool: \
`$ git pr show -t 21637`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21637.diff">https://git.openjdk.org/jdk/pull/21637.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21637#issuecomment-2429324522)

### CODE REVIEW

**Review by dfuch — APPROVED** [review]
Changes to java.management, java.naming, and sctp look good to me.

**Comment by jaikiran** [comment]
Hello Hannes, since `@param` requires a parameter name and since javadoc already warns about incorrect `@param` names (that don't match the name of the method parameter), does the order in which the `@param` is listed in the javadoc text matter? I looked at https://bugs.openjdk.org/browse/JDK-8279931 but that doesn't tell why the order would be important.

**Comment by hns** [comment]
@jaikiran it does not matter for the generated documentation because we already [put the tags in the correct order][JDK-8234682]. So it's purely a code hygiene issue to make doc comments easier to read and maintain. 

[JDK-8234682]: https://bugs.openjdk.org/browse/JDK-8234682

**Comment by hns** [comment]
> Changes to java.management, java.naming, and sctp look good to me.

Thanks @dfuch, setting reviewers to 2 for changes in the the remaining modules.

/reviewers 2

File: src/java.xml/share/classes/javax/xml/validation/SchemaFactoryConfigurationError.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
```
Comment (jaikiran) [inline src/java.xml/share/classes/javax/xml/validation/SchemaFactoryConfigurationError.java:2]: Hello Hans, this looks like an oversight - the copyright year update should have retained the 2013 and additionally introduced 2024.

Comment (hns) [reply src/java.xml/share/classes/javax/xml/validation/SchemaFactoryConfigurationError.java:2]: Thanks for catching this, @jaikiran!

**Review by jaikiran — APPROVED** [review]
This doc-only change which merely reorders the `@param` tags looks OK to me.

**Review by irisclark — APPROVED** [review]
Confirmed changes only re-order @param declarations.
 
Recommend @JoeWang-Java to review changes to java.xml since portions of that module are maintained in an upstream repository.

**Review by JoeWang-Java — APPROVED** [review]
Thanks Iris for the reminder. Changes to java.xml look good.

**Review by jddarcy — APPROVED** [review]
Changes in java.compiler look fine.

**Comment by hns** [comment]
Thanks all!

/integrate

**Pull request closed without merging**